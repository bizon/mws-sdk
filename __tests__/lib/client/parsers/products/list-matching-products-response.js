const parseXml = require('../../../../../lib/client/parsers')
const parseListMatchingProductsResponse = require('../../../../../lib/client/parsers/products/list-matching-products-response')

describe('lib.client.parsers.products.list-matching-products-response', () => {
  it('should parse the ListMatchingProductsResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <ListMatchingProductsResponse xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01">
        <ListMatchingProductsResult>
          <Products xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01"
            xmlns:ns2="http://mws.amazonservices.com/schema/Products/2011-10-01/default.xsd">
            <Product>
              <Identifiers>
                <MarketplaceASIN>
                  <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
                  <ASIN>059035342X</ASIN>
                </MarketplaceASIN>
              </Identifiers>
              <AttributeSets>
                <ns2:ItemAttributes xml:lang="en-US">
                  <ns2:Author>Rowling, J.K.</ns2:Author>
                  <ns2:Binding>Paperback</ns2:Binding>
                  <ns2:Brand>Scholastic Press</ns2:Brand>
                  <ns2:Creator Role="Illustrator">GrandPrÃ©, Mary</ns2:Creator>
                  <ns2:Edition>1st</ns2:Edition>
                  <ns2:Feature>Recommended Age: 9 years and up</ns2:Feature>
                  <ns2:ItemDimensions>
                    <ns2:Height Units="inches">0.80</ns2:Height>
                    <ns2:Length Units="inches">7.50</ns2:Length>
                    <ns2:Width Units="inches">5.20</ns2:Width>
                    <ns2:Weight Units="pounds">0.50</ns2:Weight>
                  </ns2:ItemDimensions>
                  <ns2:IsAutographed>false</ns2:IsAutographed>
                  <ns2:IsMemorabilia>false</ns2:IsMemorabilia>
                  <ns2:Label>Scholastic Paperbacks</ns2:Label>
                  <ns2:Languages>
                    <ns2:Language>
                      <ns2:Name>english</ns2:Name>
                      <ns2:Type>Unknown</ns2:Type>
                    </ns2:Language>
                    <ns2:Language>
                      <ns2:Name>english</ns2:Name>
                      <ns2:Type>Original Language</ns2:Type>
                    </ns2:Language>
                    <ns2:Language>
                      <ns2:Name>english</ns2:Name>
                      <ns2:Type>Published</ns2:Type>
                    </ns2:Language>
                  </ns2:Languages>
                  <ns2:ListPrice>
                    <ns2:Amount>10.99</ns2:Amount>
                    <ns2:CurrencyCode>USD</ns2:CurrencyCode>
                  </ns2:ListPrice>
                  <ns2:Manufacturer>Scholastic Paperbacks</ns2:Manufacturer>
                  <ns2:NumberOfItems>1</ns2:NumberOfItems>
                  <ns2:NumberOfPages>320</ns2:NumberOfPages>
                  <ns2:PackageDimensions>
                    <ns2:Height Units="inches">1.00</ns2:Height>
                    <ns2:Length Units="inches">7.50</ns2:Length>
                    <ns2:Width Units="inches">5.20</ns2:Width>
                    <ns2:Weight Units="pounds">0.50</ns2:Weight>
                  </ns2:PackageDimensions>
                  <ns2:PackageQuantity>1</ns2:PackageQuantity>
                  <ns2:PartNumber>9780590353427</ns2:PartNumber>
                  <ns2:ProductGroup>Book</ns2:ProductGroup>
                  <ns2:ProductTypeName>ABIS_BOOK</ns2:ProductTypeName>
                  <ns2:PublicationDate>1999-10-01</ns2:PublicationDate>
                  <ns2:Publisher>Scholastic Paperbacks</ns2:Publisher>
                  <ns2:ReleaseDate>1999-09-08</ns2:ReleaseDate>
                  <ns2:SmallImage>
                    <ns2:URL>
                      http://ecx.images-amazon.com/images/I/51MU5VilKpL._SL75_.jpg
                    </ns2:URL>
                    <ns2:Height Units="pixels">75</ns2:Height>
                    <ns2:Width Units="pixels">51</ns2:Width>
                  </ns2:SmallImage>
                  <ns2:Studio>Scholastic Paperbacks</ns2:Studio>
                  <ns2:Title>Harry Potter and the Sorcerer's Stone (Book 1)</ns2:Title>
                </ns2:ItemAttributes>
              </AttributeSets>
              <Relationships />
              <SalesRankings>
                <SalesRank>
                  <ProductCategoryId>book_display_on_website</ProductCategoryId>
                  <Rank>401</Rank>
                </SalesRank>
                <SalesRank>
                  <ProductCategoryId>15356791</ProductCategoryId>
                  <Rank>5</Rank>
                </SalesRank>
                <SalesRank>
                  <ProductCategoryId>3153</ProductCategoryId>
                  <Rank>8</Rank>
                </SalesRank>
                <SalesRank>
                  <ProductCategoryId>17468</ProductCategoryId>
                  <Rank>16</Rank>
                </SalesRank>
              </SalesRankings>
            </Product>
          </Products>
        </ListMatchingProductsResult>
        <ResponseMetadata>
          <RequestId>3b805a12-689a-4367-ba86-EXAMPLE91c0b</RequestId>
        </ResponseMetadata>
      </ListMatchingProductsResponse>`
    )

    const res = parseListMatchingProductsResponse(
      '/products:ListMatchingProductsResponse',
      doc
    )

    expect(res).toMatchSnapshot()
  })
})
