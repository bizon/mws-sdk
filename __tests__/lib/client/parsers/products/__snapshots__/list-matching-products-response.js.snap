// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.products.list-matching-products-response should parse the ListMatchingProducts example response from MWS doc 1`] = `
Object {
  "listMatchingProductsResult": Object {
    "products": Array [
      Object {
        "attributes": Object {
          "authors": Array [
            "Rowling, J.K.",
          ],
          "binding": "Paperback",
          "brand": "Scholastic Press",
          "creators": Array [
            Object {
              "name": "GrandPrÃ©, Mary",
              "role": "Illustrator",
            },
          ],
          "edition": "1st",
          "features": Array [
            "Recommended Age: 9 years and up",
          ],
          "isAutographed": false,
          "isMemorabilia": false,
          "itemDimensions": Object {
            "height": Object {
              "unit": "inches",
              "value": 0.8,
            },
            "length": Object {
              "unit": "inches",
              "value": 7.5,
            },
            "weight": Object {
              "unit": "pounds",
              "value": 0.5,
            },
            "width": Object {
              "unit": "inches",
              "value": 5.2,
            },
          },
          "label": "Scholastic Paperbacks",
          "languages": Array [
            Object {
              "audioFormat": null,
              "name": "english",
              "type": "Unknown",
            },
            Object {
              "audioFormat": null,
              "name": "english",
              "type": "Original Language",
            },
            Object {
              "audioFormat": null,
              "name": "english",
              "type": "Published",
            },
          ],
          "listPrice": Object {
            "amount": 10.99,
            "currencyCode": "USD",
          },
          "manufacturer": "Scholastic Paperbacks",
          "numberOfItems": 1,
          "numberOfPages": 320,
          "packageDimensions": Object {
            "height": Object {
              "unit": "inches",
              "value": 1,
            },
            "length": Object {
              "unit": "inches",
              "value": 7.5,
            },
            "weight": Object {
              "unit": "pounds",
              "value": 0.5,
            },
            "width": Object {
              "unit": "inches",
              "value": 5.2,
            },
          },
          "packageQuantity": 1,
          "partNumber": "9780590353427",
          "productGroup": "Book",
          "productTypeName": "ABIS_BOOK",
          "publicationDate": 1999-10-01T00:00:00.000Z,
          "publisher": "Scholastic Paperbacks",
          "releaseDate": 1999-09-08T00:00:00.000Z,
          "smallImage": Object {
            "height": Object {
              "unit": "pixels",
              "value": 75,
            },
            "url": "http://ecx.images-amazon.com/images/I/51MU5VilKpL._SL75_.jpg",
            "width": Object {
              "unit": "pixels",
              "value": 51,
            },
          },
          "studio": "Scholastic Paperbacks",
          "title": "Harry Potter and the Sorcerer's Stone (Book 1)",
        },
        "identifiers": Object {
          "marketplaceAsin": Object {
            "asin": "059035342X",
            "marketplaceId": "ATVPDKIKX0DER",
          },
          "skuIdentifier": null,
        },
        "offers": Array [],
        "relationships": Object {
          "variationChildren": Array [],
          "variationParent": null,
        },
        "salesRankings": Array [
          Object {
            "productCategoryId": "book_display_on_website",
            "rank": 401,
          },
          Object {
            "productCategoryId": "15356791",
            "rank": 5,
          },
          Object {
            "productCategoryId": "3153",
            "rank": 8,
          },
          Object {
            "productCategoryId": "17468",
            "rank": 16,
          },
        ],
      },
    ],
  },
  "responseMetadata": Object {
    "requestId": "3b805a12-689a-4367-ba86-EXAMPLE91c0b",
  },
}
`;
